BuildMacOSUniversalBinary.py

```diff
@@ -0,0 +1,278 @@
+    "x86_64_mac_os_deployment_target": "10.12.0"
+
+}
+# # END CONFIG # #
+
+# Architectures to build for. This is explicity left out of the command line
+# config options for several reasons:
+# 1) Adding new architectures will generally require more code changes
+# 2) Single architecture builds should utilize the normal generated cmake
+#    project files rather than this wrapper script
+
+ARCHITECTURES = ["x86_64", "arm64"]
+
+
+def parse_args(default_conf=DEFAULT_CONFIG):
+    """
+    Parses the command line arguments into a config dictionary.
+    """
+
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument(
+        '--target',
+        help='Build target in generated project files',
+        default=default_conf["build_target"],
+        dest="build_target")
+
+    parser.add_argument(
+        '--dst_app',
+        help='Directory where universal binary will be stored',
+        default=default_conf["dst_app"])
+
+    parser.add_argument(
+        '--entitlements',
+        help='Path to .entitlements file for code signing',
+        default=default_conf["entitlements"])
+
+    parser.add_argument(
+        '--codesign',
+        help='Code signing identity to use to sign the applications',
+        default=default_conf["codesign_identity"],
+        dest="codesign_identity")
+
+    for arch in ARCHITECTURES:
+        parser.add_argument(
+             '--{}_pkg_config'.format(arch),
+             help="Folder containing .pc files for {} libraries".format(arch),
+             default=default_conf[arch+"_pkg_config_path"],
+             dest=arch+"_pkg_config_path")
+
+        parser.add_argument(
+             '--{}_qt5_path'.format(arch),
+             help="Install path for {} qt5 libraries".format(arch),
+             default=default_conf[arch+"_qt5_path"])
+
+        parser.add_argument(
+             '--{}_mac_os_deployment_target'.format(arch),
+             help="Deployment architecture for {} slice".format(arch),
+             default=default_conf[arch+"_mac_os_deployment_target"])
+
+    return vars(parser.parse_args())
+
+
+def lipo(path0, path1, dst):
+    if subprocess.call(['lipo', '-create', '-output', dst, path0, path1]) != 0:
+        print("WARNING: {} and {} can not be lipo'd, keeping {}"
+              .format(path0, path1, path0))
+
+        shutil.copy(path0, dst)
+
+
+def recursiveMergeBinaries(src0, src1, dst):
+    """
+    Merges two build trees together for different architectures into a single
+    universal binary.
+
+    The rules for merging are:
+
+    1) Files that exist in either src tree are copied into the dst tree
+    2) Files that exist in both trees and are identical are copied over
+       unmodified
+    3) Files that exist in both trees and are non-identical are lipo'd
+    4) Symlinks are created in the destination tree to mirror the hierarchy in
+       the source trees
+    """
+
+    # loop over all files in src0
```

Comment:
The for statement performs globing on all files in src0, so this comment seems useless to me as it doesn't provides any insight regarding what the code is doing.

Reply:
Okay, I can remove it.